src/main/java/duke/ListManager.java

```diff
@@ -0,0 +1,150 @@
+    private static final String MESSAGE_SPACER = ". ";
+
+    private final ArrayList<Task> list;
+
+    public ListManager(ArrayList<Task> list){
+        this.list = list;
+    }
+
+    public void printList(){
+        System.out.println(MESSAGE_LIST_TASK);
+        for(int i = 0; i < list.size(); i++){
+            int itemIndex= i + 1;
+            System.out.println(itemIndex + MESSAGE_SPACER + list.get(i).toString());
+        }
+    }
+
+    public void printToDo() throws CommandException{
+        boolean haveToDo = false;
+        for(int i = 1; i <= list.size(); i++){
+            if(list.get(i) instanceof ToDo){
+                haveToDo = true;
+                break;
+            }
+        }
+        if(!haveToDo){
+            throw new CommandException(ErrorList.ERROR_EMPTY_TODO_LIST);
+        }
+        for(int i = 0; i < list.size(); i++){
+            int itemIndex= i + 1;
+            if(list.get(i) instanceof ToDo ) {
+                System.out.println(itemIndex + MESSAGE_SPACER + list.get(i).toString());
+            }
+        }
+    }
+
+    public void printEvent() throws CommandException{
+        boolean haveEvent = false;
+        for(int i = 1; i <= list.size(); i++){
+            if(list.get(i) instanceof Event) {
+                haveEvent = true;
+                break;
+            }
+        }
+        if(!haveEvent){
+            throw new CommandException(ErrorList.ERROR_EMPTY_EVENT_LIST);
+        }
+        for(int i = 0; i < list.size(); i++){
+            int itemIndex= i + 1;
+            if(list.get(i) instanceof Event ) {
+                System.out.println(itemIndex + MESSAGE_SPACER + list.get(i).toString());
+            }
+        }
+    }
+
+    public void printDeadline() throws CommandException{
+        boolean haveDeadline = false;
+        for(int i = 1; i <= list.size(); i++){
+            if(list.get(i) instanceof Deadline) {
+                haveDeadline = true;
+                break;
+            }
+        }
+        if(!haveDeadline){
+            throw new CommandException(ErrorList.ERROR_EMPTY_DEADLINE_LIST);
+        }
+        for(int i = 0; i < list.size(); i++){
+            int itemIndex= i + 1;
+            if(list.get(i) instanceof Deadline ) {
+                System.out.println(itemIndex + MESSAGE_SPACER + list.get(i).toString());
+            }
+        }
+    }
+
+    private void printAddItem(Task t){
+        System.out.println(MESSAGE_TASK_ADDED + System.lineSeparator() + t.toString() + System.lineSeparator() + MESSAGE_TASK_NOW
+                            + list.size() + MESSAGE_TASK_IN_LIST);
+    }
+
+    public void addTodo(String description, boolean isFromFile){
+        Task t = new ToDo(description);
+        list.add(t);
+        if(!isFromFile) {
+            System.out.println(Logo.dividerWithoutNewLine);
+            printAddItem(t);
+            System.out.println(Logo.dividerWithoutNewLine);
+        }
```

Comment:
perhaps improve understandability by apply the SLAP concept here? e.g a method to print these 3 lines of code?